### ISSUE TITLE
Add timeout settings (default to 5 minutes)

### ISSUE THREAD

**Issue by dadoonet** [elastic/elasticsearch#15833]
We need to backport #15080 in 2.x branch.

By default, azure does not timeout. This commit adds support for a timeout settings which defaults to 5 minutes.
It's a timeout **per request** not a global timeout for a snapshot request.

It can be defined globally, per account or both. Defaults to `5m`.

```yml
cloud:
    azure:
        storage:
            timeout: 10s
            my_account1:
                account: your_azure_storage_account1
                key: your_azure_storage_key1
                default: true
            my_account2:
                account: your_azure_storage_account2
                key: your_azure_storage_key2
                timeout: 30s
```

In this example, timeout will be 10s for `my_account1` and 30s for `my_account2`.

Related to https://github.com/elastic/elasticsearch/issues/10564#issuecomment-169690968.

**Comment by craigwi** [comment]
in case anyone needs a super short term solution, I implemented the most basic solution (the default 5min timeout) in https://github.com/craigwi/elasticsearch-cloud-azure/releases/tag/v2.8.3-craigwi.

Craig.

**Comment by dadoonet** [comment]
Closed by #15950
